python/benchmarks/basic_functions.py

```diff
@@ -78,20 +77,87 @@ def get_fresh_lib(self):
+    def peakmem_write(self, rows):
+        self.fresh_lib.write(f"sym", self.df)
 
-    def time_write_short_wide(self, rows, num_symbols):
+    def time_write_short_wide(self, rows):
         self.fresh_lib.write("short_wide_sym", self.df_short_wide)
 
-    def peakmem_write_short_wide(self, rows, num_symbols):
+    def peakmem_write_short_wide(self, rows):
         self.fresh_lib.write("short_wide_sym", self.df_short_wide)
 
+    def time_read(self, rows):
+        self.lib.read(f"sym").data
+
+    def peakmem_read(self, rows):
+        self.lib.read(f"sym").data
+
+    def time_read_short_wide(self, rows):
+        lib = self.ac[get_prewritten_lib_name(BasicFunctions.WIDE_DF_ROWS)]
+        lib.read("short_wide_sym").data
+
+    def peakmem_read_short_wide(self, rows):
+        lib = self.ac[get_prewritten_lib_name(BasicFunctions.WIDE_DF_ROWS)]
+        lib.read("short_wide_sym").data
+
+    def time_read_with_columns(self, rows):
+        COLS = ["value"]
+        self.lib.read(f"sym", columns=COLS).data
+
+    def peakmem_read_with_columns(self, rows):
+        COLS = ["value"]
+        self.lib.read(f"sym", columns=COLS).data
+
+    def time_read_with_date_ranges(self, rows):
+        self.lib.read(f"sym", date_range=BasicFunctions.DATE_RANGE).data
+
+    def peakmem_read_with_date_ranges(self, rows):
+        self.lib.read(f"sym", date_range=BasicFunctions.DATE_RANGE).data
+
+
+class BatchBasicFunctions:
+    number = 5
+    timeout = 6000
+    CONNECTION_STRING = "lmdb://batch_basic_functions?map_size=20GB"
+    DATE_RANGE = pd.date_range("2023-01-01", "2023-01-01")
+    params = BATCH_PARAMS
+    param_names = BATCH_PARAM_NAMES
+    def setup_cache(self):
+        self.ac = Arctic(BatchBasicFunctions.CONNECTION_STRING)
+        rows_values, num_symbols_values = BatchBasicFunctions.params
+
+        self.dfs = {rows: generate_pseudo_random_dataframe(rows) for rows in rows_values}
+        for rows in rows_values:
+            lib = get_prewritten_lib_name(rows)
+            self.ac.delete_library(lib)
+            self.ac.create_library(lib)
+            lib = self.ac[lib]
+            for sym in range(num_symbols_values[-1]):
+                lib.write(f"{sym}_sym", self.dfs[rows])
+
+    def teardown(self, rows, num_symbols):
+        for lib in self.ac.list_libraries():
+            if "prewritten" in lib:
+                continue
+            self.ac.delete_library(lib)
+
+        del self.ac
+
+    def setup(self, rows, num_symbols):
+        self.ac = Arctic(BatchBasicFunctions.CONNECTION_STRING)
+        self.read_reqs = [ReadRequest(f"{sym}_sym") for sym in range(num_symbols)]
+
+        self.df = generate_pseudo_random_dataframe(rows)
+        self.lib = self.ac[get_prewritten_lib_name(rows)]
+        self.fresh_lib = self.get_fresh_lib()
+
+    def get_fresh_lib(self):
+        self.ac.delete_library("fresh_lib")
+        return self.ac.create_library("fresh_lib")
+
     def time_write_staged(self, rows, num_symbols):
```

Comment:
Staged writes aren't a batch method, move to `BasicFunctions` and make sure there are multiple incomplete segments to compact

Reply:
Agreed. Thanks for flagging!
Should be fixed now.